app/screens/apps_form/apps_form_field.tsx

```diff
@@ -49,49 +49,39 @@ const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
 });
 
 export default class AppsFormField extends React.PureComponent<Props, State> {
-    constructor(props:Props) {
+    constructor(props: Props) {
         super(props);
 
-        this.state = {};
-    }
-
-    handleAutocompleteSelect = (selected: DialogOption) => {
-        if (!selected) {
-            return;
+        let selected;
+        switch (props.field.type) {
+        case AppFieldTypes.STATIC_SELECT:
+        case AppFieldTypes.DYNAMIC_SELECT:
+        case AppFieldTypes.USER:
+        case AppFieldTypes.CHANNEL: {
+            const value = props.value as AppSelectOption[] | null;
+            if (value) {
+                selected = value.map((option) => {
+                    return {
+                        text: option.label,
+                        value: option.value,
+                    };
+                });
+            }
+        }
         }
-        const {
-            field,
-        } = this.props;
-
-        this.setState({selected});
 
-        const selectedOption = {
-            label: selected.text,
-            value: selected.value,
+        this.state = {
+            selected,
         };
+    }
 
-        this.props.onChange(field.name, selectedOption);
-    };
-
-    handleMultioptionAutocompleteSelect = (selected: DialogOption[]) => {
+    handleAutocompleteSelect = (selected: DialogOption[]) => {
```

Comment:
handleAutocompleteSelect is now simple.  It only changes sets the select value which is passed to `AutocompleteSelect`